src/socket/address.cr

```diff
@@ -132,10 +132,6 @@ class Socket
       end
     end
 
-    private def address(addr) : Nil
-      # shouldn't happen
-    end
```

Comment:
I'm not sure. Either `@addr4` or `@addr6` *must* have been set in `#initialize` (otherwise it raises). The compiler can't know about it because that depends on the third-party `@family` variable. Let's avoid an overload for an impossible case?